lib/semian/activerecord_trilogy_adapter.rb

```diff
@@ -82,18 +82,25 @@ def with_resource_timeout(temp_timeout)
 
     private
 
-    def acquire_semian_resource(**)
-      super
-    rescue ActiveRecord::StatementInvalid => error
-      if error.cause.is_a?(Trilogy::TimeoutError)
-        semian_resource.mark_failed(error)
-        error.semian_identifier = semian_identifier
+    # As of Rails 7.1, TimeoutErrors are translated to ActiveRecord::AdapterTimeout
+    if ActiveRecord.version < Gem::Version.new("7.1.0.alpha")
```

Comment:
Do we really care about supporting 7.0?

Reply:
Yeah probably not, I forgot that Trilogy is only available as of Rails 7.1 anyways 😅